src/AdornedRulerPanel.cpp

```diff
@@ -1282,9 +1282,9 @@ AdornedRulerPanel::AdornedRulerPanel(AudacityProject* project,
 
    mOuter = GetClientRect();
 
-   mRuler.SetUpdater( std::make_unique<LinearUpdater>( mRuler, mViewInfo ), mLeftOffset );
+   mRuler.SetUpdater( std::make_unique<LinearUpdater>( mViewInfo ), mLeftOffset );
```

Comment:
I see that the second commit corrects several things I found doubtful in the first commit.

Good to correct them:  but better not to introduce them though at any point in the history.

Maybe the first two commits should be squashed (after rebbasing on a separate small commit for AudacityHeaders.h)

But then, make the separate, and prior, commit to define RulerStruct with all necessary changes in Ruler.cpp.  And then, rebase all onto that.

This will present a more reviewable sequence of changes.  Ruler.cpp will make a lot of necessary, "noisy" changes in a prior commit that is mostly just the noise.

Then the later commit separates the "noise" from the more interesting structural change.